plugins/virtual_event/services/VirtualEventService.php

```diff
@@ -137,4 +141,53 @@ public function listAction(KalturaVirtualEventFilter $filter = null, KalturaFilt
 		return $filter->getListResponse($pager, $this->getResponseProfile());
 	}
 	
+	protected function validateScheduleEvents (KalturaVirtualEvent $virtualEvent)
+	{
+		if (!is_null($virtualEvent->agendaScheduleEventId))
+		{
+			$this->getScheduleEvent($virtualEvent->agendaScheduleEventId, VirtualScheduleEventSubType::AGENDA);
+		}
+		if (!is_null($virtualEvent->registrationScheduleEventId))
+		{
+			$this->getScheduleEvent($virtualEvent->registrationScheduleEventId, VirtualScheduleEventSubType::REGISTRATION);
+		}
+		if (!is_null($virtualEvent->mainEventScheduleEventId))
+		{
+			$this->getScheduleEvent($virtualEvent->mainEventScheduleEventId, VirtualScheduleEventSubType::MAIN_EVENT);
+		}
+	}
+	
+	protected function validateUserGroups (KalturaVirtualEvent $virtualEvent)
+	{
+		if (!is_null($virtualEvent->adminsGroupId))
+		{
+			$this->getGroup($virtualEvent->adminsGroupId);
+		}
+		if (!is_null($virtualEvent->attendeesGroupId))
+		{
+			$this->getGroup($virtualEvent->attendeesGroupId);
+		}
+	}
+	
+	protected function getGroup($groupId)
+	{
+		$dbGroup = kuserPeer::getKuserByPartnerAndUid($this->getPartnerId(), $groupId);
+		if(!$dbGroup || $dbGroup->getType() != KuserType::GROUP)
+		{
+			throw new KalturaAPIException(KalturaGroupErrors::INVALID_GROUP_ID, $groupId);
+		}
+	}
+	
+	protected function getScheduleEvent($eventId, $subType)
```

Comment:
check if there ia a function that returns by id and partner on the peer and use it, if not move the function to schedule peer

Reply:
you dont use the returned value so change the function name, get is misleading